src/commonMain/kotlin/com/charleskorn/kaml/YamlInput.kt

```diff
@@ -267,7 +267,7 @@ private sealed class YamlMapLikeInputBase(map: YamlMap, context: SerializersModu
     protected fun <T> fromCurrentValue(action: YamlInput.() -> T): T {
         try {
             return action(currentValueDecoder)
-        } catch (e: YamlException) {
```

Comment:
Should this stay as `YamlException`?

Reply:
Based on the current code only `SinglePathYamlException` can be thrown here. But it probably would be more future–proof to catch 'em all and do more checks inside. But if you'd go with not traversing the whole tree the root exception could stay the same so this change would be reverted.